### PR TITLE
Proof of stake snark build fixes

### PR DESCRIPTION
**Pull request by nholland94**
Fixed 3 bugs in the proof of stake for snark builds:
1) ocaml implementation wasn't properly tracking epoch length
2) the negative one and genesis epoch lengths was wrong
3) seed update logic in snark was checking the current slot, not the previous slot

During investigation, I also added more rigorous tests for `in_seed_update_range_var`.

### CODE REVIEW

File: src/lib/consensus/proof_of_stake.ml [hunk 1]
```diff
@@ -253,9 +253,14 @@ module Make (Inputs : Inputs_intf) : Intf.S = struct
...
 
       let%test_unit "in_seed_update_range unchecked vs. checked equality" =
-        Quickcheck.test ~trials:100 gen ~f:(fun slot ->
-            Test_util.test_equal Unpacked.typ Snark_params.Tick.Boolean.typ
-              in_seed_update_range_var in_seed_update_range slot )
+        let test =
+          Test_util.test_equal Unpacked.typ Snark_params.Tick.Boolean.typ
+            in_seed_update_range_var in_seed_update_range
+        in
+        Quickcheck.test ~trials:100 gen ~f:test ;
```
Comment (bkase) [inline src/lib/consensus/proof_of_stake.ml]: quickcheck has an optional arg for edge cases `~examples`

File: src/lib/consensus/proof_of_stake.ml [hunk 2]
```diff
@@ -786,7 +791,7 @@ module Make (Inputs : Inputs_intf) : Intf.S = struct
           , epoch_length ) )
       in
       let curr_seed, curr_lock_checkpoint =
-        if Epoch.Slot.in_seed_update_range curr_slot then
+        if Epoch.Slot.in_seed_update_range prev_slot then
```
Comment (bkase) [inline src/lib/consensus/proof_of_stake.ml]: oops
